quinn/src/send_stream.rs

```diff
@@ -43,53 +43,80 @@ impl SendStream {
         }
     }
 
-    /// Write bytes to the stream
+    /// Attempts to write a buffer into this stream, returning how many bytes were written
     ///
-    /// Yields the number of bytes written on success. Congestion and flow control may cause this to
-    /// be shorter than `buf.len()`, indicating that only a prefix of `buf` was written.
+    /// Unless this method errors, it waits until some amount of `buf` can be written into this
+    /// stream, and then writes as much as it can without waiting again. Due to congestion and flow
+    /// control, this may be shorter than `buf.len()`. On success this resolves to the length of
+    /// the prefix that was written.
     ///
-    /// This operation is cancel-safe.
+    /// This method is cancellation safe: If it does not resolve, no bytes were written.
     pub async fn write(&mut self, buf: &[u8]) -> Result<usize, WriteError> {
         Write { stream: self, buf }.await
     }
 
-    /// Convenience method to write an entire buffer to the stream
+    /// Attempts to write an buffer into this stream in its entirety
     ///
-    /// This operation is *not* cancel-safe.
+    /// This method repeatedly calls [`write`](Self::write) until all bytes are written, or an
+    /// error occurs.
+    ///
+    /// This method is *not* cancellation safe: Even if this does not resolve, some prefix of `buf`
+    /// may have been written when previously polled.
     pub async fn write_all(&mut self, buf: &[u8]) -> Result<(), WriteError> {
         WriteAll { stream: self, buf }.await
     }
 
-    /// Write chunks to the stream
+    /// Like [`write`](Self::write), except that it writes from a slice of [`Bytes`]
```

Comment:
I don't like these formulations for the one-line description; I think they're relying too much on indirection. It could be useful to explain the functionality in relation to `Self::write()`, but not in the first line.